safehttp/status.go

```diff
@@ -19,10 +19,13 @@ package safehttp
 type StatusCode int
 
 const (
-	StatusOK                  StatusCode = 200 // RFC 7231, 6.3.1
-	StatusMovedPermanently    StatusCode = 301 // RFC 7231, 6.4.2
-	StatusBadRequest          StatusCode = 400 // RFC 7231, 6.5.1
-	StatusUnauthorized        StatusCode = 401 // RFC 7231, 3.1
-	StatusForbidden           StatusCode = 403 // RFC 7231, 6.5.3
-	StatusInternalServerError StatusCode = 500 // RFC 7231, 6.6.1
+	StatusOK                   StatusCode = 200 // RFC 7231, 6.3.1
+	StatusNoContent            StatusCode = 204 // RFC 7231, 6.3.5
+	StatusMovedPermanently     StatusCode = 301 // RFC 7231, 6.4.2
+	StatusBadRequest           StatusCode = 400 // RFC 7231, 6.5.1
+	StatusUnauthorized         StatusCode = 401 // RFC 7231, 3.1
+	StatusForbidden            StatusCode = 403 // RFC 7231, 6.5.3
+	StatusMethodNotAllowed     StatusCode = 405 // RFC 7231, 6.5.5
+	StatusUnsupportedMediaType StatusCode = 415 // RFC 7231, 6.5.13. RFC 7694, 3.
+	StatusInternalServerError  StatusCode = 500 // RFC 7231, 6.6.1
```

Comment:
Specify the type just once, it's going to be the same for the rest of the block.

Reply:
This actually doesn't seem to work, because when I specify the constants like this:
```go
const (
	StatusOK                   StatusCode = 200 // RFC 7231, 6.3.1
	StatusNoContent                       = 204 // RFC 7231, 6.3.5
       // and so on ...
}
```

My tests fail saying that:
```
safehttp/plugins/collector/collector_test.go:177:63: invalid operation: got != want (mismatched types safehttp.StatusCode and int)
safehttp/plugins/collector/collector_test.go:327:63: invalid operation: got != want (mismatched types safehttp.StatusCode and int)
```
The tests in `hsts_test` and `staticheaders_test` also fails in the same way.

When I hover the `StatusXxxx` constant in vscode it seems like it is an `untyped int` and not a `StatusCode`
![image](https://user-images.githubusercontent.com/9297883/91810248-dc2f6b00-ec2d-11ea-9534-a7cf40cdb786.png)

Reply:
Yeah, I got surprised by that yesterday. :| Sorry, it seems to work only when you do `iota`.